cpp/Contact.cpp

```diff
@@ -370,11 +393,12 @@ void dolfinx_contact::Contact::assemble_vector(
 
   // Select which side of the contact interface to loop from and get the
   // correct map
-  auto active_facets = _cell_facet_pairs[origin_meshtag];
-  auto map = _facet_maps[origin_meshtag];
-  auto facet_map = _submeshes[_opposites[origin_meshtag]].facet_map();
-  auto parent_cells = _submeshes[_opposites[origin_meshtag]].parent_cells();
-  std::size_t max_links = std::max(_max_links[0], _max_links[1]);
+  auto active_facets = _cell_facet_pairs[_contact_pairs[pair][0]];
```

Comment:
We really need to stop using auto now that the fetching of variables are quite complex.
Here I suggest you first get a reference to `const std::array<int, 2>& contact_pair = _contact_pairs[pair]`, as it is used in the following lines, then get the first or second entry to get `const std::vector<std::pair<std::int32_t, int>>& active_facets = _cell_facet_pairs[contact_pair[0]]` etc.